### PR TITLE
Added contexts for logger

### PR DESCRIPTION
**Pull request by michael1011**
This pr adds contexts to the logger as in #99.

But I am not sure about the fact that every context has its own static instance in the `Logger` class. What about a constructor that takes only the context as argument? `logDir` and `level` are the same anyway.

### CODE REVIEW

File: lib/Logger.ts [hunk 1]
```diff
@@ -13,7 +13,13 @@ enum Level {
...
+  GLOBAL = 'GLOBAL',
+  DB = 'DB',
+  RPC = 'RPC',
+  P2P = 'P2P',
+  ORDERBOOK = 'ORDERBOOK',
+  LND = 'LND',
+  RAIDEN = 'RAIDEN',
 }
 
 const contextFileMap = {
```
Comment (sangaman) [inline lib/Logger.ts]: I think you want to update `contextFileMap` too to have all the new contexts, even though for now they will all point to a single log file.

**Comment by michael1011** [comment]
Isn't it possible to use an `enum` as parameter? Something like:
```
constructor(context: Context) {
```

**Comment by sangaman** [comment]
Yeah I figure an enum would work. I guess one other advantage with static loggers is that you'll never use more than 1 instance of `Logger` per context. Is there a downside to how it's currently set up? Personally I don't have much preference either way.

**Comment by michael1011** [comment]
I just think that
```

  public static global = new Logger({ context: Context.GLOBAL, logDir: undefined, level: undefined });
  public static db = new Logger({ context: Context.DB, logDir: undefined, level: undefined });
  public static rpc = new Logger({ context: Context.RPC, logDir: undefined, level: undefined });
  public static p2p = new Logger({ context: Context.P2P, logDir: undefined, level: undefined });
  public static orderbook = new Logger({ context: Context.ORDERBOOK, logDir: undefined, level: undefined });
  public static lnd = new Logger({ context: Context.LND, logDir: undefined, level: undefined });
  public static raiden = new Logger({ context: Context.RAIDEN, logDir: undefined, level: undefined });
```
isn't the most elegant way solving this.

**Comment by moshababo** [comment]
@michael1011 Thanks!

I added a small fix for removing unnecessary props from the instantiation.

**Pull request merged**